packages/flutter/lib/src/gestures/mouse_cursor.dart

```diff
@@ -0,0 +1,332 @@
+  ///
+  /// All parameters must not be null.
+  const MouseCursorActivateDetails({
+    @required this.device,
+    @required this.platformDelegate,
+  }) : assert(device != null), assert(platformDelegate != null);
+
+  /// The pointer device that should change cursor.
+  final int device;
+
+  /// The delegate of the platform that the program is currently running on,
+  /// with which the cursor can perform operations related to mouse cursor.
+  final MouseCursorPlatformDelegate platformDelegate;
+}
+
+/// An interface for mouse cursor definitions.
+///
+/// A mouse cursor is a graphical image on the screen that echoes the movement
+/// of a pointer device, such as a mouse or a stylus. A [MouseCursor] object
+/// is a stateless definition of a kind of mouse cursor, such as an arrow,
+/// a pointing hand, or an I-beam.
+///
+/// A [MouseCursor] object is used by being assigned to a region. The most
+/// common way of doing so is [MouseRegion], which is wrapped and exposed by
+/// many other widgets.
+///
+/// When a pointer enters a region that is assigned with a [MouseCursor], the
+/// cursor's [MouseCursor.activate] is called.
+///
+/// See also:
+///
+///  * [MouseRegion], which is a common way of assigning a region with a
+///    [MouseCursor].
+///  * [MouseTracker], which determines the cursor that each device should show,
+///    and dispatches the changing callbacks.
+///  * [SystemMouseCursors], which provies many system cursors.
+///  * [NoopMouseCursors], which is a special type of mouse cursor that does
+///    nothing.
+@immutable
+abstract class MouseCursor {
+  /// Create a mouse cursor.
+  ///
+  /// The base constructor does nothing.
+  const MouseCursor();
+
+  /// Change a pointer device to this cursor based on the information provided
+  /// by `details`.
+  ///
+  /// The `details` contains the target device and the platform delegate. The
+  /// cursor can control the platform using `delegate.platformDelegate`.
+  ///
+  /// This method resolves to `true` if the operation is successful, `false` if
+  /// the operation is unsupported by the platform, or rejects to error if the
+  /// operation is implemented but an error occurs.
+  ///
+  /// This method is called by [MouseTracker] when a mouse pointer enters a
+  /// region that is assigned with this mouse cursor.
+  Future<bool> activate(MouseCursorActivateDetails details);
+
+  /// A platform-independent short description for this cursor.
+  ///
+  /// It is usually one or several readable words.
+  String describeCursor();
+
+  @override
+  String toString() {
+    return '$runtimeType(${describeCursor()})';
+  }
+}
+
+/// A mouse cursor that does nothing when activated.
+///
+/// See also:
+///
+///  * [SystemMouseCursors.releaseControl], which is an object of this cursor.
+@immutable
+class NoopMouseCursor extends MouseCursor {
+  /// Create a [NoopMouseCursor].
+  const NoopMouseCursor();
+
+  /// Does nothing and immediately returns true.
```

Comment:
we generally don't want to override the docs of virtual methods, it hides the documentation of the superclass. better to put that information in the class docs.